hollow/src/main/java/com/netflix/hollow/core/read/engine/HollowReadStateEngine.java

```diff
@@ -200,6 +201,43 @@ public Map<String, Long> calcApproxShardSizePerType() {
         return typeShardSizes;
     }
 
+    /* *
+     * Finds the top-level types in the Hollow dataset. A top-level type is one that is not a
+     * reference from another type, i.e., it is not an element of a list, set, or map, nor is it
+     * a field in an object schema.
+     *
+     * @return a set of top-level type names
+     */
+    public Set<String> topLevelTypes() {
```

Comment:
consider moving this to a HollowSchemaUtil in `com.netflix.hollow.core.schema`

Reply:
Done.